pkg/api/latest/codec.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package latest
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/conversion"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/serializer/json"
+	"k8s.io/kubernetes/pkg/runtime/serializer/recognizer"
+)
+
+var (
```

Comment:
Yes I agree as well - I need to break the code that depends on global
codecs today, and identify where codec is an injection vs global.  There
are a few places where universal decoder is all that is all that is needed,
but injection is still important.




On Dec 1, 2015, at 7:07 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/api/latest/codec.go
<https://github.com/kubernetes/kubernetes/pull/17922#discussion_r46359926>:

> +package latest
> +
> +import (
> +	"fmt"
> +	"io"
> +	"reflect"
> +
> +	"k8s.io/kubernetes/pkg/api"
> +	"k8s.io/kubernetes/pkg/api/unversioned"
> +	"k8s.io/kubernetes/pkg/conversion"
> +	"k8s.io/kubernetes/pkg/runtime"
> +	"k8s.io/kubernetes/pkg/runtime/serializer/json"
> +	"k8s.io/kubernetes/pkg/runtime/serializer/recognizer"
> +)
> +
> +var (

Recently, every time I've written code with global variables like this,
I've later regretted it and replaced with some containing object. It makes
testing easier, construction more well defined, etc. Up to you though.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17922/files#r46359926>.